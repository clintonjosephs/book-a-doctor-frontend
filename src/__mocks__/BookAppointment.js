import React from 'react';

function BookAppointment() {
  return (
    <div>
      <section>
        <div>
          <h1>Book an Appointment</h1>
          <form>
            <select aria-label="doctor_name">
              <option value="1">Meredith Grey</option>
              <option value="2">Cristina Yang</option>
              <option value="3">George O Malley</option>
              <option value="4">Izzie Stevens</option>
              <option value="5">Alex Karev</option>
            </select>
            <input aria-label="city" disabled="" type="text" value="city" readOnly />
            <input id="date_appointment" required="" type="date" readOnly />
            <button type="submit">Reserve</button>
          </form>
        </div>
      </section>
    </div>
  );
}

export default BookAppointment;
