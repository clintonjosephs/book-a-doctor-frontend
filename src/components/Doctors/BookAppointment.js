import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { addAppointmentDispatcher, fetchAllDoctors } from '../../redux/bookDoctor/doctorThunks';
import styles from './BookAppointment.module.css';

function BookAppointment() {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const currentDoctor = useSelector((state) => state.bookDoctorReducer.doctor[0] || null);
  const [dateAppointment, setDateAppointment] = useState('');
  const [selectedDoctorcity, setSelectedDoctorcity] = useState('');
  const [selectedDoctor, setSelectedDoctor] = useState('');
  let doctors = useSelector((state) => state.bookDoctorReducer.doctors.data || []);

  useEffect(() => {
    if (currentDoctor) {
      setSelectedDoctorcity(currentDoctor.city);
      setSelectedDoctor(currentDoctor.id);
    }
    dispatch(fetchAllDoctors());
  }, [dispatch]);

  const handleChange = (e) => {
    setSelectedDoctorcity(doctors[e.target.value].city);
    setSelectedDoctor(doctors[e.target.value].id);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addAppointmentDispatcher(selectedDoctor, dateAppointment));
    navigate('/myreservations');
  };

  if (currentDoctor) {
    doctors = [currentDoctor];
  }

  return (
    <section className={styles.book_appointment_container}>
      <div className={styles.book_appointment}>
        <h1 className={styles.h1}>Book an Appointment </h1>
        <form onSubmit={(e) => handleSubmit(e)}>
          <select
            onChange={(e) => handleChange(e)}
            className={styles.inputs}
            aria-label="doctor_name"
          >
            {doctors.map((doctor, index) => (
              <option key={doctor.id} value={index}>
                {doctor.name}
              </option>
            ))}
          </select>
          <input
            type="text"
            value={selectedDoctorcity}
            disabled
            className={styles.disabled}
            aria-label="city"
          />

          <input
            type="date"
            onChange={(event) => {
              setDateAppointment(event.target.value);
            }}
            required
            className={styles.inputs}
            id="date_appointment"
          />
          <button type="submit">Reserve</button>
        </form>
      </div>
    </section>
  );
}

export default BookAppointment;
