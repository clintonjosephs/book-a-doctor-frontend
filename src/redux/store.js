import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import bookDoctorReducer from './bookDoctor/doctorActions';
import userReducer from './user/userActions';
import reservationReducers from './reservation/reservationActions';

const reducer = combineReducers({
  bookDoctorReducer,
  userReducer,
  reservationReducers,
});

// creating a store
const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
