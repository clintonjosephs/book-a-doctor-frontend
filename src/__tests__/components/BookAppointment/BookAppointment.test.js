import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import BookAppointment from '../../../components/Doctors/BookAppointment';
import store from '../../../redux/store';

it('BookAppointment renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <BookAppointment />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
