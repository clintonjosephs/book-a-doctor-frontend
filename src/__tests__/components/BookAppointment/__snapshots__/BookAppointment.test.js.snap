// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookAppointment renders correctly 1`] = `
<section
  className="book_appointment_container"
>
  <div
    className="book_appointment"
  >
    <h1
      className="h1"
    >
      Book an Appointment 
    </h1>
    <form
      onSubmit={[Function]}
    >
      <select
        aria-label="doctor_name"
        className="inputs"
        onChange={[Function]}
      />
      <input
        aria-label="city"
        className="disabled"
        disabled={true}
        type="text"
        value=""
      />
      <input
        className="inputs"
        id="date_appointment"
        onChange={[Function]}
        required={true}
        type="date"
      />
      <button
        type="submit"
      >
        Reserve
      </button>
    </form>
  </div>
</section>
`;
